src/main/java/ca/dmoj/java/DisallowedClassRule.java

```diff
@@ -0,0 +1,23 @@
+package ca.dmoj.java;
+
+public class DisallowedClassRule {
+    protected interface ExceptionFactory {
+        Exception create();
+    }
+
+    protected String disallowedClassName;
+    protected ExceptionFactory disallowedClassException;
```

Comment:
You can probably just call this `className` and `exception`. Same for accessors.